src/App.tsx

```diff
@@ -14,210 +34,76 @@ export const App: React.FC = () => {
     }
   }, []);
 
+  const loadTodos = useCallback(async () => {
+    if (!user) {
+      return;
+    }
+
+    try {
+      setErrorMessage('');
+      const loadedTodos = await getTodos(user.id);
+
+      setTodos(loadedTodos);
+    } catch (error) {
+      setErrorMessage('Something went wrong while loading todos');
+    }
+  }, []);
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter(todo => {
+      switch (filterType) {
+        case FilterType.All:
+          return todo;
```

Comment:
AFAIS filter todo for FilterType.All just make a copy or todos. 
In addition to my comment about the switch case, If you need a copy make `[...todos]`, if you don't need just return original `todos`